core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -78,7 +76,7 @@ class GroupMetadataManager(brokerId: Int,
   private val shuttingDown = new AtomicBoolean(false)
 
   /* number of partitions for the consumer metadata topic */
-  private val groupMetadataTopicPartitionCount = getGroupMetadataTopicPartitionCount
+  private var groupMetadataTopicPartitionCount: Int = _
```

Comment:
Wonder if this needs `@volatile`.  It is only set once, upon invoking `startup()`, at which time it changes from 0 to the partition count.  The only place this value is used is in `partitionFor()` where it is the denominator and will cause division-by-zero if the original value is read instead of the updated value.

Reply:
If it's used by multiple threads and not set in the constructor, then it needs to be volatile. If it's only set in the constructor and there is at least one final field, then it would not need volatile.